core/imt/inc/ROOT/RTaskArena.hxx

```diff
@@ -0,0 +1,79 @@
+// @(#)root/thread:$Id$
+// // Author: Xavier Valls Pla   08/05/20
+//
+/*************************************************************************
+ * Copyright (C) 1995-2020, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+//////////////////////////////////////////////////////////////////////////
+//                                                                      //
+// RTaskArena                                                           //
+//                                                                      //
+// This file implements the method to initialize and retrieve ROOT's    //
+// global task arena, together with a method to check for active        //
+// CPU bandwith control, and a class to wrap the tbb task arena with    //
+// the purpose of keeping tbb off the installed headers                 //
+//                                                                      //
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef ROOT_RTaskArena
+#define ROOT_RTaskArena
+
+#include "RConfigure.h"
+#include <memory>
+
+// exclude in case ROOT does not have IMT support
+#ifndef R__USE_IMT
+// No need to error out for dictionaries.
+# if !defined(__ROOTCLING__) && !defined(G__DICTIONARY)
+#  error "Cannot use ROOT::Internal::RTaskArenaWrapper without defining R__USE_IMT."
+# endif
+#else
+
+/// tbb::task_arena is an alias of tbb::interface7::task_arena, which doesn't allow
+/// to forward declare tbb::task_arena without forward declaring tbb::interface7
+namespace tbb{
+namespace interface7{class task_arena;}
+using task_arena = interface7::task_arena;
+}
```

Comment:
Yes, that was discussed in the previous PPP, it's in the commit message, and I tried to solve it in two different PRs unsuccessfully. Didn't find a way to keep at least a forward declaration of tbb::task_arena out of the headers because we need tbb::task_arena::execute() in TThreadExecutor.cxx, which implies the definition of the method.

On the other hand, tbb::task_arena is an alias to tbb::interface7::task_arena and  you cannot forward declare aliases https://stackoverflow.com/questions/18834645/c-forward-declaration-and-alias-with-using-or-typedef

Reply:
I would just #include the corresponding TBB header and have a smoother TBB upgrade next time or allowing people to use different versions of TBB with ROOT. But as you said, it was discussed in the PPP meeting.

Reply:
I understand, but  the requirement here was keeping TBB out of ROOT headers